packages/dashboard/src/util/resizeSelectionBox.spec.ts

```diff
@@ -90,15 +97,60 @@ describe('should not horizontally resize selection box below minimum height', ()
   topAnchors.forEach((anchor) => {
     const vector = { x: 10, y: 10 };
     it(`should not resize selection box on ${anchor} anchor`, () => {
-      expect(resizeSelectionBox({ selectionBox: curr, anchor, vector }).height).toEqual(2);
+      expect(resizeSelectionBox({ selectionBox: curr, anchor, vector, grid }).height).toEqual(2);
     });
   });
 
   const bottomAnchors: Anchor[] = ['bottom-left', 'bottom-right', 'bottom'];
   bottomAnchors.forEach((anchor) => {
     const vector = { x: 10, y: -10 };
     it(`should not resize selection box on ${anchor} anchor`, () => {
-      expect(resizeSelectionBox({ selectionBox: curr, anchor, vector }).height).toEqual(2);
+      expect(resizeSelectionBox({ selectionBox: curr, anchor, vector, grid }).height).toEqual(2);
+    });
+  });
+});
+const withInGrid = (newRect: Rect, gridRect: Rect) => {
+  const { x, y, width, height } = newRect;
+  const { x: gridX, y: gridY, width: gridWidth, height: gridHeight } = gridRect;
+  return x >= gridX && y >= gridY && x + width <= gridX + gridWidth && y + height <= gridY + gridHeight;
+};
+describe('should not resize selection box beyond grid', () => {
+  const anchors: Anchor[] = ['top', 'top-left', 'top-right', 'bottom', 'bottom-left', 'bottom-right', 'left', 'right'];
+  const startingPoints = [
```

Comment:
use the `test.each` instead of rolling your own way of specifying collection of test caseds

Reply:
updated